include/lsst/meas/extensions/simpleShape.h

```diff
@@ -23,47 +23,90 @@
-public:
+class SimpleShapeResult;
 
+/**
+ *  @brief A C++ control class to handle SdssShapeAlgorithm's configuration
+ *
+ */
+class SimpleShapeControl {
+ public:
     LSST_CONTROL_FIELD(sigma, double, "Sigma of circular Gaussian used as weight function, in pixels");
     LSST_CONTROL_FIELD(nSigmaRegion, double, "Maximum radius for pixels to include, in units of sigma");
 
-    SimpleShapeControl() :
-        algorithms::ShapeControl("shape.simple"), sigma(1.5), nSigmaRegion(3) {}
-
-private:
-    virtual PTR(algorithms::AlgorithmControl) _clone() const;
-    virtual PTR(algorithms::Algorithm) _makeAlgorithm(
-        afw::table::Schema & schema, PTR(daf::base::PropertyList) const & metadata
-    ) const;
+    SimpleShapeControl() : sigma(1.5), nSigmaRegion(3) {}
 };
 
-/**
- *  Struct to hold the results of SimpleShape when we don't run it as a plugin.
- */
-struct SimpleShapeResult {
-    afw::geom::ellipses::Quadrupole ellipse; ///< Measured second moments.
-    afw::geom::Point2D center; ///< Measured first moments, or the input center if !recentroid
-    Eigen::Matrix<double,5,5> covariance; ///< Matrix of uncertainties; ordered Ixx, Iyy, Ixy, Ix, Iy.
+class SimpleShapeResultKey : public afw::table::FunctorKey<SimpleShapeResult> {
+ public:
+    static SimpleShapeResultKey addFields(
+            afw::table::Schema & schema,
+            std::string const & name
+    );
+
+    // Default constructor
+    SimpleShapeResultKey() {}
+
+     /**
+     *  @brief Construct from a subschema, assuming _xx, _yy, etc. subfields
+     *
+     *  If a schema has "a_xx", "a_yy", etc. fields, this constructor allows you to construct
+     *  a SimpleShapeResultKey via:
+     *  @code
+     *  SimpleShapeResultKey k(schema["a"]);
+     *  @endcode
+     */
+    SimpleShapeResultKey(lsst::afw::table::SubSchema const & s);
+
+    virtual SimpleShapeResult get(afw::table::BaseRecord  const & record) const;
+    virtual void set(afw::table::BaseRecord & record, SimpleShapeResult const & value) const;
+
+    //@{
+    /// Compare the FunctorKey fro equality with another, using the underlying Keys
+    bool operator==(SimpleShapeResultKey const & other) const;
+    bool operator!=(SimpleShapeResultKey const & other) const { return !(*this == other); }
+    //@}
+
+    /// Return True if the key is valid
+    bool isValid() const;
+
+    lsst::meas::base::FlagHandler const & getFlagHandler() const { return _flagHandler; }
+
+ private:
+    lsst::afw::table::QuadrupoleKey _shapeResult;
+    lsst::afw::table::Point2DKey _centroidResult;
+    lsst::afw::table::CovarianceMatrixKey<double, 5 > _uncertantyResult;
+    lsst::meas::base::FlagHandler _flagHandler;
 };
 
-class SimpleShape : public algorithms::ShapeAlgorithm {
-public:
+
+class SimpleShape : public lsst::meas::base::SimpleAlgorithm {
+ public:
 
     typedef SimpleShapeControl Control;
     typedef SimpleShapeResult Result;
+    typedef SimpleShapeResultKey ResultKey;
```

Comment:
I suggest removing the above two typedefs. They aren't provided by any other measurement plugin (that I could find) and I think they aren't very useful. In this .h and associated .cc file it's clearer to use the full name, and users will rarely need either because they can usually use `auto`.